course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala

```diff
@@ -419,6 +422,82 @@ class CourseEnrolmentActor @Inject()(@Named("course-batch-notification-actor") c
         } else 0}
         enrolmentData.setStatus(getCompletionStatus(enrolmentData.getProgress, leafNodesCount))
     }
+
+    def enrollProgram(request: Request): Unit = {
+        val programId: String = request.get(JsonKey.PROGRAM_ID).asInstanceOf[String]
+        val userId: String = request.get(JsonKey.USER_ID).asInstanceOf[String]
+        val batchId: String = request.get(JsonKey.BATCH_ID).asInstanceOf[String]
+        val batchData: CourseBatch = courseBatchDao.readById(programId, batchId, request.getRequestContext)
+        val enrolmentData: UserCourses = userCoursesDao.read(request.getRequestContext, userId, programId, batchId)
+        val batchUserData: BatchUser = batchUserDao.read(request.getRequestContext, batchId, userId)
+        validateEnrolment(batchData, enrolmentData, true)
+        val contentData = ContentUtil.getContentReadV2(programId, request.getContext.getOrDefault(JsonKey.HEADER, new util.HashMap[String, String]).asInstanceOf[util.Map[String, String]])
+        if (contentData.size() == 0 || !util.Arrays.asList(getConfigValue(JsonKey.ENROLL_COLLECTION_ALLOW_PRIMARY_CATEGORY).split(","): _*).contains(contentData.get(JsonKey.PRIMARYCATEGORY).asInstanceOf[String]))
+            ProjectCommonException.throwClientErrorException(ResponseCode.accessDeniedToEnrolOrUnenrolCourse, programId);
```

Comment:
Move this right after reading programId from request
so that if content is not available we no need to call 3 tables unnecessarily.

Reply:
Done